Red_Black_Tree/RedBlackTree.java

```diff
@@ -0,0 +1,331 @@
+import java.util.*;
+/*
+RED BLACK TREE IN JAVA
+*/
+
+
+public class RedBlackTree{
+    public static int BLACK=1;
+    public static int RED=0;
+    public static class Node{
+        int data;
+        int color;
+        Node left;
+        Node right;
+        Node parent;
+        boolean isNull;
+    }
+    public static Node createNode(int data, int color, Node parent)
+    {
+        Node node=new Node();
+        node.data=data;
+        node.color=color;
+        node.left=nullNode(node);
+        node.isNull=false;
+        node.right=nullNode(node);
+        if(parent!=null)
+        {
+            node.parent=parent;
```

Comment:
Add a space around all binary operators except " . "
Never add space around unary operators, increment (++) and decrement (--) operators.